rules/ast/ast.go

```diff
@@ -749,55 +756,185 @@ func labelsEqual(labels1, labels2 clientmodel.Metric) bool {
 // Eval implements the VectorNode interface and returns the result of
 // the expression.
 func (node *VectorArithExpr) Eval(timestamp clientmodel.Timestamp) Vector {
-	result := Vector{}
-	if node.lhs.Type() == ScalarType && node.rhs.Type() == VectorType {
-		lhs := node.lhs.(ScalarNode).Eval(timestamp)
+	// vector-to-vector operation
```

Comment:
Style consistency: All comments in Prometheus should be sentence-style (start with capital letter and end with period). Dito everywhere else.